### ISSUE TITLE
Bug: generated `*.typegen.ts` ignoring eslint rules 

### ISSUE THREAD

**Issue by martynchamberlin** [statelyai/xstate#3791]
### Description

When using the typegen syntax, e.g.:

```ts
  tsTypes: {} as import('./machine.typegen').Typegen0,
```

The resultant `machine.typegen.ts` file does not take into account our `.eslintrc` rules. Maybe calling this a _bug_ is a bit heavy-handed, but it seems like it'd be a good thing to support. 

For context, we aren't keeping our `typegen.ts` files in `gitignore` because we have a large team and we're introducing these state machines into a narrow part of the codebase for now. Other team members who aren't working on parts of the codebase containing xstate are having to generate these files each time they grab the latest version of our master branch, whenever the state machine coded has changed, since TS evaluates a codebase as a whole, including parts that might not be actually included in a given WebPack entry. We considered making this a postinstall hook for npm, but having to run `npm install` every time you grab latest master is also annoying. So tl;dr we decided to keep these files in version control. But that is highlighting the fact that they're not getting lint-ified, which is causing its own set of issues, with our complex tooling setup. Don't want to sound all negative though — loving xstate so far!

### Expected result

```ts
// This file was automatically generated. Edits will be overwritten

export interface Typegen0 {
  '@@xstate/typegen': true;
  internalEvents: {
    'done.invoke.(machine):invocation[0]': { type: 'done.invoke.(machine):invocation[0]'; data: unknown; __tip: 'See the XState TS docs to learn how to strongly type this.' };
'xstate.init': { type: 'xstate.init' };
  };
  invokeSrcNameMap: {
    'myService': 'done.invoke.(machine):invocation[0]';
  };
  missingImplementations: {
    actions: never;
    delays: never;
    guards: never;
    services: never;
  };
  eventsCausingActions: {
    'consoleLogId': 'done.invoke.(machine):invocation[0]';
  };
  eventsCausingDelays: {

  };
  eventsCausingGuards: {

  };
  eventsCausingServices: {
    'myService': 'xstate.init';
  };
  matchesStates: undefined;
  tags: never;
}
```

### Actual result

```ts

  // This file was automatically generated. Edits will be overwritten

  export interface Typegen0 {
        '@@xstate/typegen': true;
        internalEvents: {
          "done.invoke.(machine):invocation[0]": { type: "done.invoke.(machine):invocation[0]"; data: unknown; __tip: "See the XState TS docs to learn how to strongly type this." };
"xstate.init": { type: "xstate.init" };
        };
        invokeSrcNameMap: {
          "myService": "done.invoke.(machine):invocation[0]";
        };
        missingImplementations: {
          actions: never;
          delays: never;
          guards: never;
          services: never;
        };
        eventsCausingActions: {
          "consoleLogId": "done.invoke.(machine):invocation[0]";
        };
        eventsCausingDelays: {

        };
        eventsCausingGuards: {

        };
        eventsCausingServices: {
          "myService": "xstate.init";
        };
        matchesStates: undefined;
        tags: never;
      }

```

### Reproduction

https://xstate.js.org/docs/guides/typescript.html#typing-promise-services

### Additional context

xstate version: 4.35.2

**Comment by Andarist** [comment]
> So tl;dr we decided to keep these files in version control. 

This is a reasonable approach to choose. I think that I personally would just always keep them in the version control - they can be quite helpful when reviewing since you might see in the diff what changed there, what new events are now leading to calling a particular action etc.

> The resultant machine.typegen.ts file does not take into account our .eslintrc rules.

Do you just mean that they are not formatted in accordance with your linting rules?

**Comment by martynchamberlin** [comment]
> Do you just mean that they are not formatted in accordance with your linting rules?

Correct. Thanks. We can (and have!) tried adding them to eslint-ignore, but we’ve still encountered a time or two where they’ve gotten inadvertently linted, which has resulted in mismatched versions / merge conflicts. Conceptually, it would be nice for all of our code stored in git to be linted, for consistency and readability. If the diffs are going to show in PRs, it’s nice to at least have them linted. Just my 2¢!

**Comment by Andarist** [comment]
It sounds like a job for a precommit hook though. We have no idea what kind of fotmatters/linters any given project might be using and we cant possibly just call those tools appropriately to auto-format those files. When using the VS Code to generate those files in the background they are often auto-formatted if your VS Code has „Format on Save” enabled

**Comment by martynchamberlin** [comment]
Hmm fair. I was thinking maybe we could run `eslint --fix` or similar on just the generated file, but maybe that's too opinionated of a command.

**Comment by Andarist** [comment]
That's exactly what u can set up on your end - just run both watchers in parallel. Note that when those typegen files are generated using the IDE (and not `@xstate/cli`) we should strive to autoformat this - but it depends on your IDE config etc. That's because sometimes we can simply ask the IDE to format a file - and it should offload that job to a given extension etc. It's capable of doing that as extensions and formatting providers "register" themselves in the IDE. 

This way we don't have to handle numerous formatters and understand how to "call" them correctly.
